pallets/automation-time/src/tests.rs

```diff
@@ -231,19 +236,92 @@ fn schedule_native_transfer_works() {
 	})
 }
 
+#[test]
+fn schedule_xcmp_works() {
+	new_test_ext(START_BLOCK_TIME).execute_with(|| {
+		let call: Vec<u8> = vec![2, 4, 5];
+		get_funds(Sibling::from(PARA_ID).into_account());
+		assert_ok!(AutomationTime::schedule_xcmp_task(
+			cumulus_pallet_xcm::Origin::SiblingParachain(PARA_ID.try_into().unwrap()).into(),
+			vec![50],
+			vec![SCHEDULED_TIME],
+			PARA_ID.try_into().unwrap(),
+			call.clone(),
+			100_000,
+		));
+		match AutomationTime::get_scheduled_tasks(SCHEDULED_TIME) {
+			None => {
+				panic!("A task should be scheduled")
+			},
+			Some(task_ids) => match AutomationTime::get_task(task_ids[0]) {
+				None => {
+					panic!("A task should exist if it was scheduled")
+				},
+				Some(task) => {
+					let expected_task = Task::<Test>::create_xcmp_task(
+						Sibling::from(PARA_ID).into_account(),
+						vec![50],
+						vec![SCHEDULED_TIME].try_into().unwrap(),
+						PARA_ID.try_into().unwrap(),
+						call.clone(),
+						100_000,
+					);
+
+					assert_eq!(task, expected_task);
+				},
+			},
+		}
+	})
+}
+
+#[test]
+fn schedule_xcmp_errors_para_id_mismatch() {
+	new_test_ext(START_BLOCK_TIME).execute_with(|| {
+		let para_id_2: u32 = 2001;
+		assert_noop!(
+			AutomationTime::schedule_xcmp_task(
+				cumulus_pallet_xcm::Origin::SiblingParachain(PARA_ID.try_into().unwrap()).into(),
```

Comment:
ahh i vaguely remember you having some t rouble with origin type conversion. What was the other origin type? was it XCM OriginType or something from Substrate? Either way, I'm curious how you got past the type conversion?

Reply:
`cumulus_pallet_xcm::Origin::SiblingParachain(PARA_ID.try_into().unwrap()).into()`  === `Native`

you can also do it another way by setting the inner and outer call.
```
let dispatch_origin = XcmOriginToCallOrigin::convert_origin(
	MultiLocation { parents: 1, interior: X1(Parachain(PARA_ID)) },
	OriginKind::SovereignAccount,
).unwrap();
let call: OuterCall = Call::schedule_xcmp_task { 
	provided_id: vec![50],
	execution_times: vec![SCHEDULED_TIME + 1],
	para_id: para_id2.try_into().unwrap(),
	call: vec![4,5,6],
}.into();
assert_ok!(call.dispatch(dispatch_origin));

Reply:
ahhh i see i see. thanks!!